app/controllers/concerns/flashable.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+module Flashable
+  extend ActiveSupport::Concern
+
+  def message_for_notice(invalid_records)
+    msg = "Uploading #{upload_params[:csv_type]} succeeded"
+    msg + (invalid_records.present? ? ' with warnings:' : '.')
+  end
+
+  def errors_for_alert(invalid_records)
+    error_list = invalid_records[0, 15].map do |record|
+      line = record.errors.delete(:base)
```

Comment:
The errors are removed during display after uploading. I could move the display errors functionality into the model, but once displayed to the user, the error information and invalid records are disposable.